packages/eslint/src/generators/init/init.ts

```diff
@@ -127,22 +88,22 @@ function initEsLint(tree: Tree, options: LinterInitOptions): GeneratorCallback {
     addTargetDefaults(tree);
   }
 
-  updateVSCodeExtensions(tree);
-
-  return !options.skipPackageJson
-    ? addDependenciesToPackageJson(
+  const tasks: GeneratorCallback[] = [];
+  if (!options.skipPackageJson) {
+    tasks.push(removeDependenciesFromPackageJson(tree, ['@nx/eslint'], []));
```

Comment:
If the purpose of removing this dependency before re-adding it was to avoid version clash if the same package is installed in `dependencies` and `devDependencies` then wouldn't we want to do the same with the other packages listed below - `eslint`, `@nx/eslint-plugin`, `@typescript-eslint/*`?

Reply:
No, there's follow-up work to use the updated `addDependenciesToPackageJson`, the util PR is pending merge: https://github.com/nrwl/nx/pull/21123

Reply:
I assume that was the reason, but I'm unsure. I just left the existing behavior for this. IMHO, we shouldn't do it for any. AFAIK, the `addDependenciesToPackageJson` function handles this though maybe differently. I haven't checked.